src/index.d.ts

```diff
@@ -0,0 +1,225 @@
+interface DropdownValue {
+  label: string
+  value: number
+}
+
+interface BarInterface {
+  data: {
+    categories: string[]
+    data: string[][]
+    count: number[][]
+    bars: string[]
+  }
+  vertical?: boolean
+  labelFormatter?: (label: string) => string
+  percent?: boolean
+  showLegend?: boolean
+  id?: string
+}
+
+interface BoxInterface {
+  className?: string
+  type?: 'slim' | 'fit'
+  loading?: boolean
+}
+
+interface ButtonInterface {
+  className?: string
+  type?: 'primary' | 'secondary' | 'neutral'
+  variant?: 'solid' | 'flat' | 'outline'
+  size?: 'M' | 'S'
+  isDisabled?: boolean
+  id?: string
+  iconPosition?: 'left' | 'right'
+  to?: string
+  href?: string
+  icon?: React.ReactChild | React.ReactChild[]
+  children?: React.ReactChild | React.ReactChild[]
+  onClick: () => void
+}
+
+interface DataTableInterface {
+  rows: string[][] | React.ReactElement[][]
+  noData?: string[]
+  headers: string[] | React.ReactElement[]
+  emptyCell?: string
+  className?: string
+  Component?: React.ReactChild | React.ReactChild[]
+  loader?: boolean
+  type?: 'table'
+  perPage?: DropdownValue
+  downloadable?: boolean
+  searchable?: boolean
+  hasPagination?: boolean
+}
+
+interface DialogInterface {
+  className?: string
+  title: string
+  children: React.ReactChild | React.ReactChild[]
+  onClose: () => void
+  variant?: 'normal' | 'form'
+  disabledOutline?: boolean
+}
+
+interface DropdownInterface {
+  isSearchable?: boolean
+  options: DropdownValue[]
+  placeholder?: string
+  hasAdd?: boolean
+  handleAdd?: () => void
+  position?: 'left'
+  isMulti?: boolean
+  handleChange: () => void
+  value: {
+    label: string
+    value: string
+    number: number
+  }
+  loadOptions: (arg0: string) => {
+    label: string
+    value: number
+  }
+  noOptionsMessage?: (arg0: string) => string
+  isClearable?: boolean
+  sideHeading?: string
+  category?: {
+    label: string
+    value: string
+    category: string
+  }
+}
+
+interface FieldWrapperInterface {
+  className?: string
+  html?: boolean
+  children: React.ReactChild | React.ReactChild[]
+  helperType?: string
+  helperText?: string
+  label?: string
+}
+
+interface InputInterface {
+  type: 'text'
```

Comment:
this can be a lot more, we can just use `HTMLInputTypeAttribute`. And this prop is optional